syntax = "proto3";

package scms_agent;
option go_package = "rpc";

// Static credentials agent
service SCMSAgent {
  rpc Auth(AuthRequest) returns (AuthResponse) {}
  rpc AuthStream(stream AuthRequest) returns (AuthStreamResponse) {}
  rpc Health(Status) returns (Status) {}
}

message AuthRequest {
  string key_id = 1;
  string signed_string = 2;
  string signature = 3;
  string datestamp = 4;
  string algorithm = 5;
  bytes ip_address = 6;
  bool is_url = 7;
}

message AuthResponse {
  bytes token = 1;
}

message AuthStreamResponse {
}

message Status {
}
